File: 이것이 취업을 위한 코딩 테스트다/그리디/만들_수_없는_금액/만들_수_없는_금액.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>

using namespace std;

int main() {
  int n;
  vector<int> coins;
  cin >> n;

  for (int i = 0; i < n; i++) {
    int coin;
    cin >> coin;

    coins.push_back(coin);
  }

  sort(coins.begin(), coins.end());

  int target = 1;

  for (int i = 0; i < n; i++) {
    if (target < coins[i]) {
      break;
    }

    target += coins[i];
  }

  cout << target;
  
  return 0;
}

File: 이것이 취업을 위한 코딩 테스트다/그리디/만들_수_없는_금액/만들_수_없는_금액.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.Scanner;

public class 만들_수_없는_금액 {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    ArrayList<Integer> coins = new ArrayList<>();
    int n = sc.nextInt();

    for (int i = 0; i < n; i++) {
      coins.add(sc.nextInt());
    }

    Collections.sort(coins);

    int target = 1;

    for (int i = 0; i < n; i++) {
      if (target < coins.get(i)) {
        break;
      }

      target += coins.get(i);
    }

    System.out.println(target);
  }
}
